core/jvm/src/test/scala/scalaz/zio/ZManagedSpec.scala

```diff
@@ -332,47 +334,73 @@ class ZManagedSpec(implicit ee: org.specs2.concurrent.ExecutionEnv) extends Test
     unsafeRun {
-      managed.timeout(1.millisecond).use(res => ZIO.succeed(res must be_===(None)))
+      for {
+        latch   <- Promise.make[Nothing, Unit]
+        managed = ZManaged.make(latch.await)(_ => ZIO.unit)
+        res     <- managed.timeout(Duration.Zero).use(res => ZIO.succeed(res must_=== (None)))
+        _       <- latch.succeed(())
+      } yield res
     }
-  }
 
-  def timeoutAcquisitionAndReservation = {
-    val managed = ZManaged(clock.sleep(1.millisecond) *> ZIO.succeed(Reservation(clock.sleep(5.seconds), ZIO.unit)))
+  private def timeoutAcquisition =
     unsafeRun {
-      managed.timeout(100.millisecond).use(res => ZIO.succeed(res must be_===(None)))
+      for {
+        latch   <- Promise.make[Nothing, Unit]
+        managed = ZManaged.reserve(Reservation(latch.await, ZIO.unit))
+        res     <- managed.timeout(Duration.Zero).use(res => ZIO.succeed(res must_=== (None)))
+        _       <- latch.succeed(())
+      } yield res
     }
-  }
 
-  def timeoutRunFinalizers = {
-    var release = 0
+  private def timeoutAcquisitionAndReservation = {
     val managed = ZManaged(
-      clock.sleep(1.millisecond) *> ZIO.succeed(Reservation(clock.sleep(5.seconds), ZIO.effectTotal(release += 1)))
+      clock.sleep(30.millisecond) *> ZIO.succeed(Reservation(clock.sleep(30.millisecond), ZIO.unit))
     )
     unsafeRun {
-      managed.timeout(100.millisecond).use(res => ZIO.succeed((res must be_===(None)) && (release must be_===(1))))
+      managed.timeout(50.millisecond).use(res => ZIO.succeed(res must be_===(None)))
     }
   }
 
-  def timed = {
+  private def timeoutRunFinalizers1 =
+    unsafeRun {
+      for {
+        latch   <- Promise.make[Nothing, Unit]
+        managed = ZManaged.reserve(Reservation(latch.await, ZIO.unit))
+        res     <- managed.timeout(Duration.Zero).use(res => ZIO.succeed(res must_=== (None)))
+        _       <- latch.succeed(())
+      } yield res
+    }
+
+  private def timeoutRunFinalizers2 =
+    unsafeRun {
+      for {
+        acquireLatch <- Promise.make[Nothing, Unit]
+        releaseLatch <- Promise.make[Nothing, Unit]
+        managed      = ZManaged(acquireLatch.await *> ZIO.succeed(Reservation(ZIO.unit, releaseLatch.succeed(()))))
+        res          <- managed.timeout(Duration.Zero).use(res => ZIO.succeed(res must_=== (None)))
+        _            <- acquireLatch.succeed(())
+        _            <- releaseLatch.await
+      } yield res
+    }
+
+  private def timed = {
     val managed = ZManaged(
-      clock.sleep(100.milliseconds) *> ZIO.succeed(Reservation(clock.sleep(100.milliseconds), ZIO.unit))
+      clock.sleep(20.milliseconds) *> ZIO.succeed(Reservation(clock.sleep(20.milliseconds), ZIO.unit))
     )
     unsafeRun {
-      managed.timed.use { case (duration, _) => ZIO.succeed((duration >= 200.milliseconds) must beTrue) }
+      managed.timed.use { case (duration, _) => ZIO.succeed(duration must be_>=(40.milliseconds)) }
```

Comment:
This one looks more reasonable since sleeping for 20 ms + 20 ms should always involve the passage of >= 40 ms of time.

_Should_, of course. 😊

Reply:
Hopefully :D. You never know with clocks

This one will probably be prone to quite a lot of false negatives though. Should we just remove it?

Reply:
I have never seen a false positive here, because scheduling + overhead will never come up _short_. You could be conservative and do `>= 39`, but I don't think you have to worry about this one.

_Think_. 😄

Reply:
I meant that if the actual code is buggy the test would have a good chance of still passing. Even if only reserve or acquire was measured, etc.

Reply:
Ah in that case, the test has marginal value, maybe best to either delete, or run it a bunch of times like a few other tests in RTSSPec.